src/lib.rs

```diff
@@ -504,6 +505,82 @@ pub use restate_sdk_macros::object;
 /// For more details, check the [`service` macro](macro@crate::service) documentation.
 pub use restate_sdk_macros::workflow;
 
+/// ### JSON Schema Generation
```

Comment:
This documentation is terrific :star_struck: 

I would move it though in the `serde` mod docs though, and just mention it few lines above when there's the link to the `Serialization` mod